### ISSUE TITLE
Adversarial Loss Not Backpropagating to Generator Due to .detach() Usage

### ISSUE THREAD

**Issue by SBeairsto** [climagination/ClimatExML#47]
**Description:**

It appears that the adversarial loss (`-torch.mean(self.C(sr))`) is currently **detached** from the computation graph in the generator’s loss function, preventing gradients from flowing through the critic and into the generator.

#### **Current code:**

`loss_g = -torch.mean(self.C(sr).detach()) + self.alpha * mean_squared_error(sr, hr)` 

#### **What happens:**

-   The  `.detach()`  on  `self.C(sr)`  isolates the adversarial term from generator backpropagation.
-   As a result,  **the generator is only updated based on the content loss**  (CRPS).

----------

#### 1.  **Gradient Inspection**
I ran an experiment comparing generator parameter gradients:

1.  **With adversarial term present (detached)**
2.  **With and without adversarial term (no detach)**

**Result:**

-   When using  `.detach()`, gradients for all generator parameters are  **identical**  whether the adversarial term is present or not.  

    _Example:_
  ```
   Layer: conv3.2.weight
      Max absolute difference: 0.00
      Mean absolute difference: 0.00
      Grad norm (with adv): 2.940111e+02
      Grad norm (without adv): 2.940111e+02
      ✓ Gradients are IDENTICAL` 
 ```
    
-   When the  `.detach()`  is  _removed_, gradients differ, showing the adversarial term influences generator updates:  

	 _Example:_
```
    Layer: conv3.2.weight
      Max absolute difference: 0.625
      Mean absolute difference: 0.129
      Grad norm (with adv): 2.721126e+02
      Grad norm (without adv): 2.668307e+01
      ✗ Gradients are DIFFERENT` 
```
	
#### 2.  **Training Curve Comparison**

-   Training runs performed with and without the adversarial loss term in the generator loss produced  **nearly identical training curves**  and metrics, further indicating the adversarial component is not influencing generator learning under the current implementation.

#### 3.  **Reference to Standard Practice**

-   I reviewed other PyTorch Lightning WGAN implementations, and in all standard examples,  `.detach()`  is  **not**  applied to  `self.C(sr)`  during generator updates. The adversarial loss is always computed as:
    `loss_g = -torch.mean(self.C(sr)) + content_loss`

----------

### **Conclusion:**

-   The current implementation  **prevents the generator from learning to “fool” the critic**  as intended in GANs.
-   To enable proper adversarial feedback,  `.detach()`  should be  **removed**  from the loss function during the generator update:

#### **Recommended fix:**
```python 
loss_g = -torch.mean(self.C(sr)) + self.alpha * mean_squared_error(sr, hr)
```
